src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceResultSetMetaData.java

```diff
@@ -124,12 +124,12 @@ public String getSchemaName(int column) throws SQLException {
     @Override
     public int getPrecision(int column) throws SQLException {
         //TODO: Check if this is needed.
-        return 9;
+        return 18;
     }
 
     @Override
     public int getScale(int column) throws SQLException {
-        return 9;
+        return 38;
```

Comment:
We would be having scale and precisions apart from 38, 18.

Even though we are supporting Decimal(38, 18) only within Data Cloud, based on the operations, the clients can actually get different scale and precision.

For example, client can simply do a cast - `SELECT CAST(COUNT(*) AS DECIMAL(10,5) FROM TABLE`. The output of the same would be having a different scale and precision.